Week2/js/HomeTask2.js

```diff
@@ -0,0 +1,250 @@
+    let clickToReadMore = document.createElement("button");
+    clickToReadMore.setAttribute("value", "Continue Reading");
+    clickToReadMore.setAttribute("class", "continue-reading");
+    clickToReadMore.setAttribute("id", i + 1);
+    clickToReadMore.addEventListener("click", showPoPUp);
+    clickToReadMore.innerText = "Continue Reading";
+    content.appendChild(title);
+    content.appendChild(description);
+    content.appendChild(clickToReadMore);
+    newsFeed.appendChild(logo);
+    newsFeed.appendChild(content);
+    divisions[i] = newsFeed;
+  }
+  return divisions;
+};
+var channels = generateItems();
+
+var createCategories = function() {
+  let selectCategory = document.createElement("select");
+  selectCategory.setAttribute("class", "dropdown");
+  selectCategory.setAttribute("id", "dropdown");
+  selectCategory.addEventListener("change", displaySelectedItem);
+  let allCategory = document.createElement("option");
+  allCategory.setAttribute("value", "ALL");
+  allCategory.innerText = "ALL";
+  selectCategory.appendChild(allCategory);
+  for (let i = 0; i < 10; i++) {
+    let category = document.createElement("option");
+    category.setAttribute("value", i + 1);
+    category.innerText = channelsData[i].name;
+    selectCategory.appendChild(category);
+  }
+  return selectCategory;
+};
+
+var displayAll = function(dataDiv) {
+  channels.forEach(element => {
+    dataDiv.appendChild(element);
+  });
+  mainDivision.appendChild(dataDiv);
+};
+
+var removeAll = function() {
+  let currentDataDiv = document.getElementById("data-div");
+  let items = currentDataDiv.childNodes;
+  var child = currentDataDiv.lastElementChild;
+  while (child) {
+    currentDataDiv.removeChild(child);
+    child = currentDataDiv.lastElementChild;
+  }
+  while (currentDataDiv.hasChildNodes()) {
+    currentDataDiv.removeChild();
+  }
+};
+
+var displaySelectedItem = function() {
+  let selectedItem = document.getElementById("dropdown").value;
+  if (selectedItem == "ALL") {
+    removeAll();
+    displayAll(dataDiv);
+  }
+  for (let i = 0; i < 10; i++) {
+    if (selectedItem == channels[i].id) {
+      removeAll();
+      dataDiv.appendChild(channels[i]);
+    }
+  }
+};
+// to show popup when continue reading button is clicked
+function showPoPUp() {
+  alert(channelsData[this.id - 1].address);
+}
+
+//This function is to validate an email Id entered by the user.
+var validateEmailId = function() {
+  let emailId = document.getElementById("email").value;
+  //regular expression to validate email id.
+  let emailRegex = /^[a-z0-9._]+@[a-z0-9]+.[a-z]{2,3}/;
+  if (emailRegex.test(emailId)) {
+    //if email id is valid then saving it to local storag.

```

Comment:
You have to reset error message here. Otherwise after successful addition also invalid email error will be there